ddtrace/contrib/anthropic/patch.py

```diff
@@ -93,29 +95,34 @@ def traced_chat_model_generate(anthropic, pin, func, instance, args, kwargs):
 
         chat_completions = func(*args, **kwargs)
 
-        if isinstance(chat_completions, anthropic.Stream) or isinstance(
-            chat_completions, anthropic.lib.streaming._messages.MessageStreamManager
+        if (
+            isinstance(chat_completions, anthropic.Stream)
+            or isinstance(chat_completions, anthropic.lib.streaming._messages.MessageStreamManager)
+            or isinstance(chat_completions, anthropic.lib.streaming._messages.AsyncMessageStreamManager)
         ):
-            pass
+            stream = True
+            return handle_streamed_response(integration, chat_completions, args, kwargs, span)
         else:
             handle_non_streamed_response(integration, chat_completions, args, kwargs, span)
     except Exception:
         span.set_exc_info(*sys.exc_info())
         raise
     finally:
-        if integration.is_pc_sampled_llmobs(span):
-            integration.llmobs_set_tags(span=span, resp=chat_completions, args=args, kwargs=kwargs)
-
-        span.finish()
+        # we don't want to finish the span if it is a stream as it will get finished once the iterator is exhausted
+        if not stream:
+            if integration.is_pc_sampled_llmobs(span):
+                integration.llmobs_set_tags(span=span, resp=chat_completions, args=args, kwargs=kwargs)
+            span.finish()
     return chat_completions
 
 
 @with_traced_module
 async def traced_async_chat_model_generate(anthropic, pin, func, instance, args, kwargs):
     chat_messages = get_argument_value(args, kwargs, 0, "messages")
     integration = anthropic._datadog_integration
+    stream = False
 
-    operation_name = func.__name__
+    operation_name = "stream" if "stream" in kwargs else func.__name__
```

Comment:
I believe it always has to be? They overload the `.create` and `.stream` methods [here](https://github.com/anthropics/anthropic-sdk-python/blob/6023c7d03ce6ae3da603ec1624f8652852f81b51/src/anthropic/resources/messages.py#L45). So i don't see how an arg instead of a kwarg could be used.

Reply:
create() and stream() do not use args, so we should be good here.